src/garmin_parser.c

```diff
@@ -1502,7 +1517,7 @@ garmin_parser_get_field (dc_parser_t *abstract, dc_field_type_t type, unsigned i
 	case DC_FIELD_ATMOSPHERIC:
 		return DC_STATUS_UNSUPPORTED;
 	case DC_FIELD_DIVEMODE:
-		return DC_STATUS_UNSUPPORTED;
+		return DC_FIELD_VALUE(garmin->cache, value, DIVEMODE);
```

Comment:
This looks fine, but that whole "garmin_parser_get_field()" thing should probably just be deleted, and it should use the standard dc_field_get() helper instead, since it uses all the normal field-cache logic.